src/Mvc/Mvc.ViewFeatures/src/RazorComponents/StaticComponentRenderer.cs

```diff
@@ -71,15 +71,15 @@ private void InitializeStandardComponentServices(HttpContext httpContext)
             {
                 _initialized = true;
 
+                var navigationManager = (IHostEnvironmentNavigationManager)httpContext.RequestServices.GetRequiredService<NavigationManager>();
```

Comment:
Was there a functional reason for moving the initialization here?

Reply:
Yes.

I was playing with the auth sample + prerendering to test on a more complex scenario and found out that the remote authentication service was throwing because it accesses navigation manager within the constructor.

It’s better to initialize the navigation manager first since that’s lower level and not really designed to be extended by customers